llvm/test/tools/llvm-dwp/X86/cu_tu_units_manual_v5_invalid.s

```diff
@@ -0,0 +1,75 @@
+# This test checks that llvm-dwarfdump correctly reports errors when parsing
+# DWARF Unit Headers in DWP files
+
+# RUN: llvm-mc -triple x86_64-unknown-linux %s -filetype=obj -o %t.o \
+# RUN:         -split-dwarf-file=%t.dwo -dwarf-version=5
+# RUN: llvm-dwp %t.dwo -o %t.dwp
+# RUN: llvm-dwarfdump -debug-info -debug-cu-index -debug-tu-index \
+# RUN:                -manaully-generate-unit-index %t.dwp 2>&1 | FileCheck %s
+
+## Note: In order to check whether the type unit index is generated
+## there is no need to add the missing DIEs for the structure type of the type unit.
+
+# CHECK-NOT: .debug_info.dwo contents:
+
+# CHECK-DAG: .debug_cu_index contents:
+# CHECK: Failed to parse CU header in DWP file
+# CHECK-NEXT: DWARF unit at offset 0x00000000 has unsupported version 6, supported are 2-5
+
+# CHECK-DAG: .debug_tu_index contents:
+# CHECK: Failed to parse CU header in DWP file
+# CHECK-NEXT: DWARF unit at offset 0x00000000 has unsupported version 6, supported are 2-5
```

Comment:
Hmm, could you add a bit more text to the checks here - I thought the idea was that this patch would lead to one error instead of two, so I'd have thought it'd all be on one line like "error: Failed to parse CU header in DWP file: DWARF unit at offset ... "

But perhaps I've misunderstood / would be good to see the whole text, maybe?

Reply:
(Oh, and I should say - thanks for working up this test, I know it's a bit of a pain to hand-craft DWARF like this!)

Reply:
As James pointed out above, my code was not actually creating one error from the two but concatenating them together. I'll update the code and the test to be closer to what we expect from it.

Reply:
Hmm - rather than having to concatenate at every place nested errors are nested - could we have the code that prints it out do the concatenation? (so the API itself has high fidelity - still keeps the separate/nested Errors, but if a caller wants to render that to a user, concatenation is probably the right strategy?)

I don't feel strongly about this - if someone approves this patch/reckons this is the right way to go as-is, I'm OK with it.

Reply:
I think that's possible but I'd rather see that done in a follow-up since it's outside the intended scope of this change.